packages/virtualized-lists/Lists/VirtualizedList.d.ts

```diff
@@ -152,7 +152,10 @@ export interface VirtualizedListWithoutRenderItemProps<ItemT>
   /**
    * Rendered in between each item, but not at the top or bottom
    */
-  ItemSeparatorComponent?: React.ComponentType<any> | null | undefined;
+  ItemSeparatorComponent?:
+    | React.ComponentType<any | {highlighted: boolean; leadingItem: ItemT}>
```

Comment:
I took a look at this, and I think we only ever pass the two props listed here. Could we instead update this to something like?

```suggestion
    | React.ComponentType<{highlighted: boolean; leadingItem: ItemT}>
```

And update `VirtualizedListProps` and `VirtualizedListCellRenderer` as well?

Reply:
I guess custom props are allowed too. We should ideally add a test for that maybe (the scenario is rendering a prop set by `seperators.updateProps`).

```suggestion
    | React.ComponentType<{highlighted: boolean; leadingItem: ItemT; [customProp: string]: any}>
```

Reply:
Thanks, updated! Does the test have already been added? https://github.com/facebook/react-native/blob/main/packages/virtualized-lists/Lists/__tests__/VirtualizedList-test.js#L232